dist/actions/entrypoint.sh

```diff
@@ -82,23 +82,25 @@ if [ -e package.json ] && [ ! -d node_modules ]; then
     npm install
 fi
 
-# Now just pass along all arguments to the Pulumi CLI.
-OUTPUT=$(sh -c "pulumi --non-interactive $*" 2>&1)
+# Now just pass along all arguments to the Pulumi CLI, sending the output to a file for
+# later use. Note that we exit immediately on failure (under set -e), so we `tee` stdout, but
+# allow errors to be surfaced in the Actions log.
+PULUMI_COMMAND="pulumi $*"
+OUTPUT_FILE="/tmp/out.txt"
+echo "#### :tropical_drink: \`$PULUMI_COMMAND\`"
+bash -c "$PULUMI_COMMAND" | tee $OUTPUT_FILE
```

Comment:
A few questions about this:

1. Was it intentional to drop the `--non-interactive` flag, if so, why?  I would expect this is only ever used in an non-interactive context.

2. Previously, because we redirected stderr to stdout, `$OUTPUT` would contain both stdout and stderr.  Now, `$OUTPUT_FILE` will only contain output to stdout, which means we will not include stderr output in the message we post back to GitHub, which doesn't seem like what we'd want?

3. I believe that `EXIT_CODE` will no longer behave as you expect, because of the the pipeline.  You should instead set EXIT_CODE to the value of `${PIPESTATUS[0]}`.  For example, here it would be incorrect to just use $?, since the bash command itself failed.

```
[matell@matell pulumi]$ bash -c "false" | tee foo.txt; echo "$? ${PIPESTATUS[0]} ${PIPESTATUS[1]}"
0 1 0
```

Reply:
> Was it intentional to drop the --non-interactive flag, if so, why? 

Yes, as it gets passed as an argument to this script: https://github.com/pulumi/pulumi/blob/c8460156433aa9a58fe7b58717a08bd4cb1bc313/dist/actions/Dockerfile#L30

> Now, $OUTPUT_FILE will only contain output to stdout, which means we will not include stderr output in the message we post back to GitHub, which doesn't seem like what we'd want?

That's intentional, but you can check my thinking: In the event of a failure of any kind (Pulumi CLI execution or otherwise), the GitHub Actions log will show that failure, which we definitely want, and would get here. What we wouldn't get (and I don't think we'd want) would be CLI _failures_ posted back to the PR as comments. (If we do want to post comments with CLI failures, we'd need to remove `set -e`.)

> I believe that EXIT_CODE will no longer behave as you expect [...]

Ah, yes -- nice catch.

Reply:
>  What we wouldn't get (and I don't think we'd want) would be CLI failures posted back to the PR as comments.

Makes sense.  I am not sure if there is any high value messages we log to `stderr` today, but as you say, it's probably fine to push folks towards the actions console to see this.

> (If we do want to post comments with CLI failures, we'd need to remove set -e, as that's the source of the issue this PR aims to fix.)

As an aside, we need not remove `set -e` in this case.  We can figure out a way to keep `set -e` (which is helpful in general) and still allow some commands to fail and have us handle the failures gracefully (without aborting the entire script) if that's what we want to do.